### PR TITLE
release/public-v2: bugfixes for building MPI, HDF5 and NetCDF

### PR DESCRIPTION
**Pull request by climbfuji**
A small number of bugfixes that we found when building across different platforms:
- build static libjpeg
- use non-mpi compilers for HDF5 (parallel builds are turned off for release/public-v2, and MPI_*_Compiler are not defined if MPI is built as part of NCEPLIBS-external)
- add missing `LIBS` argument for building the NetCDF C library
- documentation update for macOS with clang+gfortran to solve OpenMP linker issue for NCEPLIBS

This has been tested on a range of platforms and is good to merge.

### CODE REVIEW

**Comment by climbfuji** [comment]
@kgerheiser do you have time to look at this? If not please let me know and I will merge as is, placing my confidence on the successful CI tests and manual installations. Thanks.

File: doc/README_macos_clanggfortran.txt [hunk 1]
```diff
@@ -101,11 +103,10 @@ the following commands should suffice to build the model.
 
 export INSTALL_PREFIX=/usr/local/NCEPLIBS-ufs-v2.0.0
 
-export CC=clang-9
-export FC=gfortran-9
-export CXX=clang++-9
-export PATH="/usr/local/opt/llvm/bin:$PATH"
-export LD_LIBRARY_PATH="/usr/local/opt/llvm/lib:$LD_LIBRARY_PATH"
+export PATH="/usr/local/opt/llvm@10/bin:$PATH"
```
Comment (kgerheiser) [inline doc/README_macos_clanggfortran.txt:106]: What is the point of these lines?

Comment (kgerheiser) [reply doc/README_macos_clanggfortran.txt:106]: But I approve

**Review by kgerheiser — APPROVED** [review]

File: doc/README_macos_clanggfortran.txt (hunk 1 above)
Comment (climbfuji) [reply doc/README_macos_clanggfortran.txt:106]: You need those lines to find the correct version of clang (the homebrew-installed LLVM clang with OpenMP support) and to link to the OpenMP library. This is also what homebrew tells you to do. Without them, cmake can't find the OpenMP C library, no matter what you do.

**Comment by climbfuji** [comment]
Thanks, @kgerheiser.

**Pull request merged**